subgraph/queries/setts.py

```diff
@@ -15,6 +16,7 @@
 thegraph_client = make_gql_client("thegraph")
 
 
+@lru_cache
```

Comment:
Why add lru cache here?

Reply:
This is because the last synced block can change as the code runs due to the subgraph indexing.C aching this means that we use the same last synced block at the start of the cycle when fetching the metadata for the end block and also when fetching the claimable data. In the future we could move this cached data to somewhere like redis.